### PR TITLE
Use reusable vulnerability check workflow

### PR DESCRIPTION
**Pull request by ymorimo**
## Description

This PR changes to use the reusable vuln-check workflow (scalar-labs/actions#9).

## Related issues and/or PRs

- scalar-labs/actions#9

## Changes made

- Replaced the vulnerability checking job In vuln-check.yaml file with a reusable workflow call.
- Removed scripts in `ci/vuln-check/`
- No changes in manual-vuln-check.yaml and scheduled-vuln-check.yaml

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

Used the new reusable vulnerability check workflow.

### CODE REVIEW

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by jnmt — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: .github/workflows/vuln-check.yaml
[the remark this answers is not in the document]
Comment (ymorimo) [reply .github/workflows/vuln-check.yaml:1]: Currently `vuln-check.yaml` has only the function not to duplicate the `with:` block.

[the remark this answers is not in the document]
Comment (ymorimo) [reply .github/workflows/vuln-check.yaml:1]: Currently, the `vuln-check.yaml` file has only the functionality to not duplicate the `with:` block. It has several repository-specific default values.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**